taiga_halo2/src/note.rs

```diff
@@ -96,15 +96,15 @@ pub struct RandomSeed([u8; 32]);
 pub struct InputNoteProvingInfo {
     pub note: Note,
     pub merkle_path: MerklePath,
-    app_vp_verifying_info: Box<dyn ValidityPredicateVerifyingInfo>,
-    app_vp_verifying_info_dynamic: Vec<Box<dyn ValidityPredicateVerifyingInfo>>,
+    app_vp_verifying_info: Box<ValidityPredicate>,
+    app_vp_verifying_info_dynamic: Vec<Box<ValidityPredicate>>,
```

Comment:
It seems here we are still essentially being reliant upon a `dyn trait`, if we were to expand this alias, it ends up being

`app_vp_verifying_info: Box<dyn ValidityPredicateVerifyingInfo>`
`app_vp_verifying_info_dynamic: Vec,Box<dyn ValidityPredicateVerifyingInfo>>`

meaning that binding to this is still hard if this is exposed as the outside interface